File: Cosmos-IOS/Cosmostation/Cosmostation/Model/Vote_V1.swift
//
//  Vote_V1.swift
//  Cosmostation
//
//  Created by 정용주 on 2021/03/07.
//  Copyright © 2021 wannabit. All rights reserved.
//

import Foundation

public struct Vote_V1 {
    public static let OPTION_YES           = "VOTE_OPTION_YES"
    public static let OPTION_NO            = "VOTE_OPTION_NO"
    public static let OPTION_VETO          = "VOTE_OPTION_NO_WITH_VETO"
    public static let OPTION_ABSTAIN       = "VOTE_OPTION_ABSTAIN"
    
    var proposal_id: String?
    var voter: String?
    var option: String?
    
    init(_ dictionary: NSDictionary?) {
        self.proposal_id = dictionary?["proposal_id"] as? String
        self.voter = dictionary?["voter"] as? String
        self.option = dictionary?["option"] as? String
    }
}

File: Cosmos-Android/app/src/main/java/wannabit/io/cosmostaion/model/Vote_V1.java
package wannabit.io.cosmostaion.model;

import com.google.gson.annotations.SerializedName;

public class Vote_V1 {
    public final static String OPTION_YES           = "VOTE_OPTION_YES";
    public final static String OPTION_NO            = "VOTE_OPTION_NO";
    public final static String OPTION_VETO          = "VOTE_OPTION_NO_WITH_VETO";
    public final static String OPTION_ABSTAIN       = "VOTE_OPTION_ABSTAIN";

    @SerializedName("proposal_id")
    public String proposal_id;

    @SerializedName("voter")
    public String voter;

    @SerializedName("option")
    public String option;

//    public enum VoteOption_V1 {
//        Unsepcified(0), Yes(1), Abstain(2), No(3), NoWithVeto(4);
//
//        VoteOption_V1(int value) {
//            value = value;
//        }
//    }

}
